NeoDebuggerUI/ViewModels/MainWindowViewModel.cs

```diff
@@ -226,20 +227,70 @@ public async Task OpenRunDialog()
 
             if (!IsSteppingOrOnBreakpoint)
             {
-                var task = modalWindow.ShowDialog();
+                var task = modalWindow.ShowDialog(new Window());
                 await Task.Run(() => task.Wait());
             }
 
             // not using getters because the properties are updated on another thread and won't update the ui
             IsSteppingOrOnBreakpoint = DebuggerStore.instance.manager.IsSteppingOrOnBreakpoint;
             ConsumedGas = DebuggerStore.instance.UsedGasCost;
 
+            EvtDebugCurrentLineChanged?.Invoke(IsSteppingOrOnBreakpoint, DebuggerStore.instance.manager.CurrentLine + 1);
             if (IsSteppingOrOnBreakpoint)
             {
                 UpdateStackPanel();
             }
         }
 
+        public void AddBreakpoint(int line)
+        {
+            DebuggerStore.instance.manager.AddBreakpoint(line - 1, SelectedFile);
+        }
+
+        public void RemoveBreakpoint(int line)
+        {
+            DebuggerStore.instance.manager.RemoveBreakpoint(line - 1, SelectedFile);
+        }
+
+        public void SetBreakpoint(int line)
+        {
+            if (Breakpoints.Contains(line))
+            {
+                // remove breakpoint
+                RemoveBreakpoint(line);
+                EvtBreakpointStateChanged?.Invoke(line, false);
+                return;
+            }
+
+            if (DebuggerStore.instance.manager.Map != null)
+            {
+                var entries = DebuggerStore.instance.manager.Map.Entries.Select(x => x.line);
+                if (entries.Contains(line))
+                {
+                    // add breakpoint in line
+                    AddBreakpoint(line);
+                    EvtBreakpointStateChanged?.Invoke(line, true);
+                    return;
+                }
+
+                try
+                {
+                    // add breakpoint in the next possible line
+                    var nextLine = entries.Where(x => x > line).Min();
+                    if (!Breakpoints.Contains(nextLine))
+                    {
+                        AddBreakpoint(nextLine);
+                        EvtBreakpointStateChanged?.Invoke(nextLine, true);
+                        return;
+                    }
+                }
+                catch
```

Comment:
What exception are you handling? Can you expecify in the catch? Which line throws the exception? Can you wrap only this specific line with try-catch?